### DISCUSSION TITLE
Optimal way to publish a package with dependencies to npm and use it with npx

### DISCUSSION THREAD

**Discussion by abensur** [google/zx#654] [General]
## Expected Behavior
Use a npm package with dependencies from npx

## Actual Behavior
`#! /usr/bin/env -S zx --install `
With this shebang it's working just fine when I run `npx etiquette`

## Steps to Reproduce the Problem
https://github.com/abensur/etiquette

## Questions
Is using `-S zx --intall` ok?
Should I bundle the dependency?
If so, do I need to bundle `zx` when using it on the shebang or only the other dependencies?

**Comment by antonmedv** [comment]
It depends.

**Reply by abensur** [reply]
Thanks for moving this into a discussion, would you mind sharing your opinion? I'm just reading the package.json and using github's CLI to star all the dependencies that have a github linked to it, however, I'm using a npm dependencie to find, if possible, the github linked to the npm package.

**Reply by antonmedv** [reply]
Just simply add deps.
